libs/wire-api/src/Wire/API/Routes/Public/Brig.hs

```diff
@@ -533,6 +539,43 @@ type UserClientAPI =
                :> "prekeys"
                :> Get '[JSON] [PrekeyId]
            )
+    :<|> NonceAPI
+
+type NonceAPI =
+  -- be aware that the order matters, if get was first, then head requests would be routed to the get handler
+  NewNonce "head-nonce" 'HEAD 200
+    :<|> NewNonce "get-nonce" 'GET 204
+
+type NewNonce name method statusCode =
+  Named
+    name
+    ( Summary "Get a new nonce for a client, specified in the response header `Replay-Nonce` as a uuidv4 in base64url encoding"
+        :> ZUser
+        :> "nonce"
+        :> "clients"
+        :> MultiVerb1
+             method
+             '[JSON]
+             (WithHeaders '[Header "Replay-Nonce" NonceHeader, Header "Cache-Control" Text] Nonce (RespondEmpty statusCode "No Content"))
+    )
+
+newtype NonceHeader = NonceHeader Nonce
+  deriving (Eq, Show)
+  deriving newtype (FromByteString, ToByteString)
+
+instance ToParamSchema NonceHeader where
+  toParamSchema _ = toParamSchema (Proxy @Text)
+
+instance ToHttpApiData NonceHeader where
+  toQueryParam nonce = cs (toByteString' nonce)
+
+instance FromHttpApiData NonceHeader where
+  parseQueryParam s =
+    first pack $ runParser parser (encodeUtf8 s)
```

Comment:
```suggestion
  deriving newtype (FromByteString, ToByteString, ToParamSchema, ToHttpApiData, FromHttpApiData)
```

...  because Nonce already has these.  If I'm missing anything important here, please add a source comment.

Reply:
We need instances of these to decode and encode base64url.